### ISSUE TITLE
 .spec.activeDeadlineSeconds parameter in k8s job overwrites default k8s behaviour 

### ISSUE THREAD

**Issue by maximenannan** [spotify/luigi#2383]
Hi there, hi @yonromai 

Recent changes in 2.7.3 in kubernetes luigi task seems to overwrite k8s default behaviour. 

In this commit https://github.com/spotify/luigi/pull/2335/commits/e27b9fb418fc3e25531e9b1d797c2915851d9429, the parameter `"activeDeadlineSeconds` has been added in `spec` to create k8s jobs.

One assumption on this parameter explained in the commit was that:
> It seems to me that it's the deadline for the Job to be scheduled. Once the Job is running, this timeout doesn't apply anymore.

But in k8s doc it's written that 
>By default, a Job will run uninterrupted unless a Pod fails, at which point the Job defers to the .spec.backoffLimit described above. Another way to terminate a Job is by setting an active deadline. Do this by setting the .spec.activeDeadlineSeconds field of the Job to a number of seconds 
https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#job-termination-and-cleanup

So it'd seem that setting `activeDeadlineSeconds` affects jobs even if they started running before the deadline

I have the feeling that the only way to disable this is to set `activeDeadlineSeconds` to a very large number, is there another way?
 
Wouldn't be nteresting to add something like 
```
if self.active_deadline_seconds is not None:
      job_json["spec"] = self.active_deadline_second
```

**Comment by kayibal** [comment]
Please fix jobs are expected now to finish in 100 seconds else they are terminated.

**Comment by yonromai** [comment]
@maximenannan sounds good to me, feel free to submit a PR.

**Comment by StasDeep** [comment]
Fix is present in 2.7.6. Can be closed now.
